File: generate_jwt.js
const crypto = require('crypto');

// IMPORTANT: Set JWT_SECRET environment variable before running
// Example: set JWT_SECRET=your-secret-key-minimum-32-characters
const secret = process.env.JWT_SECRET;

if (!secret) {
    console.error("ERROR: JWT_SECRET environment variable is required.");
    console.error("Set it with: set JWT_SECRET=your-secret-key-minimum-32-characters");
    process.exit(1);
}

if (secret.length < 32) {
    console.error("ERROR: JWT_SECRET must be at least 32 characters long.");
    process.exit(1);
}

// Header
const header = {
    alg: "HS256",
    typ: "JWT"
};

// Payload - 1 año de duración
const now = Math.floor(Date.now() / 1000);
const oneYear = 365 * 24 * 60 * 60;
const exp = now + oneYear;

const payload = {
    nameid: process.env.JWT_USER_ID || "65812A2A-2B92-4357-A48F-E3C9FD5755BE",
    email: process.env.JWT_EMAIL || "admin@gmail.com",
    unique_name: process.env.JWT_USERNAME || "Admin",
    family_name: process.env.JWT_FAMILY_NAME || "Administrator",
    nbf: now,
    exp: exp,
    iat: now
};

// Encode to base64url
function base64urlEncode(str) {
    return Buffer.from(JSON.stringify(str))
        .toString('base64')
        .replace(/\+/g, '-')
        .replace(/\//g, '_')
        .replace(/=/g, '');
}

const headerEncoded = base64urlEncode(header);
const payloadEncoded = base64urlEncode(payload);

// Create signature
const signatureInput = `${headerEncoded}.${payloadEncoded}`;
const signature = crypto
    .createHmac('sha256', secret)
    .update(signatureInput)
    .digest('base64')
    .replace(/\+/g, '-')
    .replace(/\//g, '_')
    .replace(/=/g, '');

// Create JWT
const jwt = `${headerEncoded}.${payloadEncoded}.${signature}`;

console.log("Token JWT (válido por 1 año):");
console.log(jwt);
console.log("\nFecha de creación:", new Date(now * 1000).toISOString());
console.log("Fecha de expiración:", new Date(exp * 1000).toISOString());

File: generate_jwt.py
import json
import base64
import hmac
import hashlib
from datetime import datetime, timedelta

# Header
header = {
    "alg": "HS256",
    "typ": "JWT"
}

# Payload - 1 año de duración
now = datetime.utcnow()
exp = now + timedelta(days=365)

payload = {
    "nameid": "65812A2A-2B92-4357-A48F-E3C9FD5755BE",
    "email": "admin@gmail.com",
    "unique_name": "Admin",
    "family_name": "Administrator",
    "nbf": int(now.timestamp()),
    "exp": int(exp.timestamp()),
    "iat": int(now.timestamp())
}

# Secret key
secret = "molloja-ecommerce-secret-key-super-secure-2025-minimum-32chars"

# Encode header and payload
def base64url_encode(data):
    json_data = json.dumps(data, separators=(',', ':')).encode('utf-8')
    return base64.urlsafe_b64encode(json_data).rstrip(b'=').decode('utf-8')

header_encoded = base64url_encode(header)
payload_encoded = base64url_encode(payload)

# Create signature
message = f"{header_encoded}.{payload_encoded}".encode('utf-8')
signature = hmac.new(secret.encode('utf-8'), message, hashlib.sha256).digest()
signature_encoded = base64.urlsafe_b64encode(signature).rstrip(b'=').decode('utf-8')

# Create JWT
jwt_token = f"{header_encoded}.{payload_encoded}.{signature_encoded}"

print(f"Token JWT (válido por 1 año):")
print(jwt_token)
print(f"\nFecha de creación: {now.isoformat()}Z")
print(f"Fecha de expiración: {exp.isoformat()}Z")
